tests/python/test_fields_builder.py

```diff
@@ -3,174 +3,276 @@
 def test_fields_builder_pointer():
-    n = 5
 
+    # Initialize common variables for tests below
+    shape_size_1d = 5
     fb1 = ti.FieldsBuilder()
     x = ti.field(ti.f32)
-    fb1.pointer(ti.i, n).place(x)
+    fb1.pointer(ti.i, shape_size_1d).place(x)
     fb1.finalize()
 
+    # [pointer] 1. Test and validation for one field assign
     @ti.kernel
-    def func1():
-        for i in range(n):
+    def assign_single_field_kernel_func():
+        for i in range(shape_size_1d):
             x[i] = i * 3
 
-    func1()
-    for i in range(n):
+    assign_single_field_kernel_func()
+    for i in range(shape_size_1d):
         assert x[i] == i * 3
 
+    # [pointer] 2. Test and validation for multiple fields assign with
+    #              range-for
     fb2 = ti.FieldsBuilder()
     y = ti.field(ti.f32)
-    fb2.pointer(ti.i, n).place(y)
+    fb2.pointer(ti.i, shape_size_1d).place(y)
     z = ti.field(ti.f32)
-    fb2.pointer(ti.i, n).place(z)
+    fb2.pointer(ti.i, shape_size_1d).place(z)
     fb2.finalize()
 
-    # test range-for
     @ti.kernel
-    def func2():
-        for i in range(n):
+    def assign_mutliple_field_kernel_func0(a=2, b=5, c=10):
+        # test range-for
+        for i in range(shape_size_1d):
             x[i] = i * 2
-        for i in range(n):
+        for i in range(shape_size_1d):
             y[i] = i + 5
-        for i in range(n):
+        for i in range(shape_size_1d):
             z[i] = i + 10
 
-    func2()
-    for i in range(n):
+    assign_mutliple_field_kernel_func0(2, 5, 10)
+    for i in range(shape_size_1d):
         assert x[i] == i * 2
         assert y[i] == i + 5
         assert z[i] == i + 10
 
-    # test struct-for
+    # [pointer] 3.  Test and validation for multiple field assign with
+    #               struct-for
     @ti.kernel
-    def func3():
+    def assign_mutliple_field_kernel_func1(a, b):
+        # test struct-for
         for i in y:
-            y[i] += 5
+            y[i] += a
         for i in z:
-            z[i] -= 5
+            z[i] -= b
 
-    func3()
-    for i in range(n):
+    assign_mutliple_field_kernel_func1(5, 5)
+    for i in range(shape_size_1d):
         assert y[i] == i + 10
         assert z[i] == i + 5
 
-    func1()
-    for i in range(n):
+    assign_single_field_kernel_func()
+    for i in range(shape_size_1d):
         assert x[i] == i * 3
 
 
+'''
+Test fields with builder destory.
+'''
+
+
 @ti.test(arch=[ti.cpu, ti.cuda, ti.vulkan])
 def test_fields_builder_destroy():
-    def A(i):
-        n = i * 10**3
+
+    # [destroy] Args of main body
+    test_sizes = [1]  # [1, 2, 3]
+    size_extend_factor = 1  # 10 ** 3
+    # note: currently only consider preicison all paltform supported,
+    # more detailed here: https://docs.taichi.graphics/lang/articles/basic/type#supported-primitive-types
+    field_types = [ti.f32, ti.i32]
+    field_sizes = [1]  # [1, 5, 10]
+
+    # [destroy] 1. test for single destroy multiple fields
```

Comment:
the function name below basically convey the same information ;). this is why we write function name like this. I don't think we need this comment as well, same as all comments above.

Reply:
Fixed